pilot/pkg/xds/ads.go

```diff
@@ -782,16 +782,12 @@ func (s *DiscoveryServer) pushConnection(con *Connection, pushEv *Event) error {
 
 	// Send pushes to all generators
 	// Each Generator is responsible for determining if the push event requires a push
-	wrl, ignoreEvents := con.pushDetails()
+	wrl := con.pushDetails()
 	for _, w := range wrl {
 		if err := s.pushXds(con, w, pushRequest); err != nil {
 			return err
 		}
 	}
-	if pushRequest.Full {
-		// Report all events for unwatched resources. Watched resources will be reported in pushXds or on ack.
-		reportAllEvents(s.StatusReporter, con.conID, pushRequest.Push.LedgerVersion, ignoreEvents)
```

Comment:
```
func reportAllEvents(s DistributionStatusCache, id, version string, ignored sets.String) {
	if s == nil {
		return
	}

       // here we only care about lds rds CDs eds, which all have been processed in `processRequest`
	for distributionType := range AllEventTypes {
		if ignored.Contains(distributionType) {
			// Skip this type
			continue
		}
		s.RegisterEvent(id, distributionType, version)
	}
}
```

Reply:
Yes, but the old `ignoreEvents` could return CDS/LDS/EDS/RDS... so we need to report them as "done". Otherwise it will be stuck waiting forever.

Reply:
arenot they are reported in processRequest

Reply:
And if you are not sure please test the wait manually. Its om to me

Reply:
They are not reported in processRequest, because there is never a request for that type. Imagine a proxy only subscribes to CDS. processRequest for LDS is never called.

This code ensures we say LDS=sent in the distribution logic.

I 100% agree the code is wonky and has poor tests. Its plausible we should look into removing it entirely. However, I don't think we should break it unless we are going to do that.

Reply:
Let me test it and give you feedback. 

>Imagine a proxy only subscribes to CDS. processRequest for LDS is never called.

To be clear, is this a normal sidecar or something that user implement customized?

Reply:
No LDS was bad example. But no RDS or EDS is possible  with envoy. Especially gateways

Reply:
```
root@kurator-linux-0001:~/istio-1.20.0/samples# kind load docker-image istio/pilot:98c924ef9daafed0e4c2c6c0a96dd9b70a5e3402 --name=istio-testing # load the image 
Image: "" with ID "sha256:f0e11648bb93d9dbcf0888f7581c9c4f0701e1bd1849d83fe1b43c2ee84ec36c" not yet present on node "istio-testing-control-plane", loading...
root@kurator-linux-0001:~/istio-1.20.0/samples# k edit deploy istiod -n istio-system # update istiod image
deployment.apps/istiod edited

root@kurator-linux-0001:~/istio-1.20.0/samples# istioctl experimental wait virtualservice httpbin.default --verbose
getting first version from chan
sync status: [{ProxyID:httpbin-797587ddc5-rmd9b.default ClusterVersion:1 ListenerVersion:1 RouteVersion:1} {ProxyID:istio-ingressgateway-6cb98cd6d4-l247j.istio-system ClusterVersion:1 ListenerVersion:1 RouteVersion:1}]
Received poll result: 6/6
Resource networking.istio.io/v1alpha3/VirtualService/default/httpbin present on 6 out of 6 configurations across 2 sidecars
```

Same result as before

Reply:
With this [commit](https://github.com/istio/istio/pull/48542/commits/b70ddbf6068249180cf56a6c4eb1b7b3d0036d40), i fixed `istioctl wait`